python/rpdk/typescript/templates/handlers.ts

```diff
@@ -88,14 +94,17 @@ class Resource extends BaseResource<ResourceModel> {
      * @param request The request object for the provisioning request passed to the implementor
      * @param callbackContext Custom context object to allow the passing through of additional
      * state or metadata between subsequent retries
+     * @param typeConfiguration Configuration data for this resource type, in the given account
+     * and region
      * @param logger Logger to proxy requests to default publishers
      */
     @handlerEvent(Action.Delete)
     public async delete(
         session: Optional<SessionProxy>,
         request: ResourceHandlerRequest<ResourceModel>,
         callbackContext: CallbackContext,
-        logger: LoggerProxy
+        logger: LoggerProxy,
+        typeConfiguration: TypeConfigurationModel,
```

Comment:
I see what you mean about changing the order. You guys are the experts on TypeScript here, but curious as to why we cannot use syntax like `typeConfiguration?: TypeConfigurationModel` here

Reply:
i also wondered why `?` or `Optional<>` wasn't used here -- technically I think it should be present on `callback` and `logger` as well as `typeConfiguration`.  possibly the types admin null already or the signature isn't needed.  but we tried to match existing patterns as much as possible since this isn't our codebase.

what we had wanted to do and did previously was to include the `typeConfiguration` _before_ the `logger`, to match the java signature.  but that is what caused the problem.

previous versions of the framework code of course call the old signature, ie `delete(session, request, callback, logger)`.

in the test failure we were seeing, that old framework code is running, making that exact call, but against a handler ^ based on the new template, expecting the new signature, so with the changes we had made earlier it thinks the 4th parameter is a `typeConfiguration` when the caller gave a `logger` there.  (the new framework called the new signature, so if you ran new versions at both places it worked fine.)

in the new code because we haven't changed any existing fields, just appended a new 5th, it quite happily accepts calls from the old framework to new code or even new framework to old code -- in the former case it calls the signature with 4 arguments, and those args match, and it passes `undefined` in for the `typeConfiguration`, which works fine at runtime since the TS type info is thrown away when converted to JS; and in the latter case it includes the new 5th argument as a typeConfiguration but old handlers only take 4 args which all match and it just ignores the 5th.